src/components/WatchListPage/WatchListPage.jsx

```diff
@@ -38,7 +44,16 @@ const WatchListPage = () => {
         <Typography variant="h4" sx={{ mb: 2 }}>
           Your watch list
         </Typography>
-        {watchedShows.length === 0 ? (
+        {isLoading ? (
+          <Box
+            display="flex"
+            justifyContent="center"
+            alignItems="center"
+            minHeight="200px"
+          >
+            <CircularProgress />
+          </Box>
+        ) : watchedShows.length === 0 ? (
```

Comment:
nested ternary operator is hard to follow. Please split it into multiple separated ternary operators or conditional rendering (using `&&`)